### ISSUE TITLE
Batchnormalization results in NaN (keras 1.0.7)

### ISSUE THREAD

**Issue by RuiShu** [fchollet/keras#3475]
Batchnormalization leads to NaN output in keras 1.0.7 (but not 1.0.6). 

Here's a sample script to reproduce the issue (keras 1.0.7, theano 0.8.2): https://gist.github.com/RuiShu/80c561a48a5523a8d0245cefde15727c

Unless I've done something wrong, help on this issue is much appreciated :)

**Comment by RaffEdwardBAH** [comment]
I have the same issue.
